source/CargoHold.cpp

```diff
@@ -595,3 +595,23 @@ int CargoHold::IllegalCargoFine() const
 
 	return totalFine;
 }
+
+
+
+// Returns the amount tons of illegal cargo.
+int CargoHold::IllegalCargoAmount() const
+{
+	int count = 0;
+
+	// Find any illegal outfits inside the cargo hold.
+	for(const auto &it : outfits)
+		if(it.first->Get("illegal") || it.first->Get("atrocity") > 0.)
+			count += it.second * it.first->Mass();
+
+	// Find any illegal mission cargo.
+	for(const auto &it : missionCargo)
+		if(it.first->IllegalCargoFine())
+			count += it.second;
```

Comment:
What about illegal passengers?

Reply:
I think eventually there could be some sort of "panic room" type outfit that conceals passengers, but for now let's just treat them like uninstalled cargo with its associated minimum.

Reply:
Illegal passengers aren't dealt with right now anyway so we can ignore this for now.